### ISSUE TITLE
Certain games error with multiple exceptions

### ISSUE THREAD

**Issue by deadair3210** [julesontheroad/NSC_BUILDER#127]
Happens no matter if I use my copy or if I use someone else's, this error about local variable 'titleid', and a error about a list being too big for an array show up again and again. Let me know if you need more info or examples

This is the console from the interface when trying to load one of the games, the main program just throws up the first error about the local variable assignment issue
![image](https://user-images.githubusercontent.com/1053504/81518637-4f6e4380-9304-11ea-97e6-b2b9c61d7dc3.png)

**Comment by julesontheroad** [comment]
You're using the interface, that's a gevent error due to the information not being exchanged properly in the websockets. I tried to silence it but it's root is on the gevent library running on the background so it's not enough to silence it on my libraries. Maybe doing more modifications in the eel library, something i started to do recently.
The error itself seems to be masterkey related in this case. Can you try with the current beta seems i see you use python?

**Comment by deadair3210** [comment]
Will try when I get home from work, I have multiple verified files that the program calls corrupted or outright fails to load so if you need more info I'll prob be able to get it

**Comment by deadair3210** [comment]
Don't know if its the same issue but I have some files that install and work fine on the Switch, but fail when attempting to load them into the Interface and get labeled as needscheck when renamed. It's a different error this time, complaining that a byte-type object was needed, not NoneType

**Comment by julesontheroad** [comment]
Just checking, do you have the latest masterkey (master_key_0a) in the keys file? I don't remember what was the last one in the template but the keys file would accept newer ones even if not specified.
I can load the information fine for that update.

0.98b x64
![098](https://user-images.githubusercontent.com/42461174/84250289-265fef00-ab0c-11ea-8327-0bd8d6281eed.PNG)

Current beta - python
![beta](https://user-images.githubusercontent.com/42461174/84250350-3c6daf80-ab0c-11ea-92a2-c62a5757dd2d.PNG)

**Comment by julesontheroad** [comment]
BTW the root issue in the second picture you showed me is that it can't decrypt the ncas. Basically:
1.- Icon, can't read nacp so "None" is returned (Nonetype) instead of the picture bytes. The picture wasn't probably on the db at that point so the fallback to Nintendo servers didn't go in place either.
2.- Couldn't read titleid so when the variable was going to be used it failed. I probably would need to define it in advance so the funtions continues and pop ups the data it can. Or maybe custom error.
3.- Can't read the nacp in the control nca.
Basically cnmt and control nca related functions are failling, but it looks similar as when it uses a masterkey that isn't in the file.

**Comment by deadair3210** [comment]
You were right on the nail about the last issue, I was missing 0a and didnt notice that I hadnt dumped the keys since the last update. Thought I had, my bad lol! Will reopen if I have any more issues with it

**Comment by deadair3210** [comment]
Thanks!

**Comment by deadair3210** [comment]
Might I suggest that maybe the program warn people if the user might be missing the required key if possible?

**Comment by julesontheroad** [comment]
Yep there's actually a function for it but I didn't decided if I wanted to check at the program start or hook a check per file.
The check per file is preferred in a way but my issue was with big multi xcis were it kinda would make sense to check all the keys needed for all the files used to make the xci. The thing is that could delay a lot the start of building the file depending on the situation.
Maybe I should just stick for now with checking for the last keys.

**Comment by deadair3210** [comment]
Maybe disable it or make it a option for the user to decide to use or not? Or only check when decrypting the file during the checker process

**Comment by deadair3210** [comment]
I'm working on multi-ing the entire switch library as a side project, so I end up using the program pretty much every day. If you need information on usecase or program flow I might be able to help
